src/vs/workbench/contrib/terminal/browser/chatTerminalCommandMirror.ts

```diff
@@ -0,0 +1,107 @@
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Disposable, ImmortalReference } from '../../../../base/common/lifecycle.js';
+import type { ITerminalCommand } from '../../../../platform/terminal/common/capabilities/capabilities.js';
+import { ITerminalInstance, ITerminalService, type IDetachedTerminalInstance } from './terminal.js';
+import { DetachedProcessInfo } from './detachedTerminal.js';
+import { IInstantiationService } from '../../../../platform/instantiation/common/instantiation.js';
+import { TerminalInstanceColorProvider } from './terminalInstance.js';
+import { TerminalLocation } from '../../../../platform/terminal/common/terminal.js';
+
+interface IDetachedTerminalCommandMirror {
+	attach(container: HTMLElement): Promise<void>;
+	renderCommand(): Promise<{ lineCount?: number } | undefined>;
+}
+
+/**
+ * Mirrors a terminal command's output into a detached terminal instance.
+ * Used in the chat terminal tool progress part to show command output for example.
+ */
+export class DetachedTerminalCommandMirror extends Disposable implements IDetachedTerminalCommandMirror {
+	private _detachedTerminal?: IDetachedTerminalInstance;
+	private _attachedContainer?: HTMLElement;
+
+	constructor(
+		private readonly _terminalInstance: ITerminalInstance,
+		private readonly _command: ITerminalCommand,
+		@ITerminalService private readonly _terminalService: ITerminalService,
+		@IInstantiationService private readonly _instantationService: IInstantiationService
+	) {
+		super();
+	}
+
+	async attach(container: HTMLElement): Promise<void> {
+		const terminal = await this._getOrCreateTerminal();
+		if (this._attachedContainer !== container) {
+			container.classList.add('chat-terminal-output-terminal');
+			terminal.attachToElement(container);
+			this._attachedContainer = container;
+		}
+	}
+
+	async renderCommand(): Promise<{ lineCount?: number } | undefined> {
+		const detached = await this._getOrCreateTerminal();
+		const raw = detached.xterm.raw;
+		if (!raw) {
+			return;
+		}
+		const vt = await this._getCommandOutputAsVT();
+		raw.reset();
+		if (!vt) {
+			return undefined;
+		}
+		if (!vt.text) {
+			return { lineCount: 0 };
+		}
+		raw.write(vt.text);
+		raw.scrollToBottom();
+		raw.refresh(0, raw.rows - 1);
+		return { lineCount: vt.lineCount };
+	}
+
+	private async _getCommandOutputAsVT(): Promise<{ text: string; lineCount: number } | undefined> {
+		const executedMarker = this._command.executedMarker;
+		const endMarker = this._command.endMarker;
+		if (!executedMarker || executedMarker.isDisposed || !endMarker || endMarker.isDisposed) {
+			return undefined;
+		}
+
+		const xterm = await this._terminalInstance.xtermReadyPromise;
```

Comment:
I think you only actually need the terminal instance's xterm instance. So you could call `await this._terminalInstance.xtermReadyPromise` before creating `DetachedTerminalCommandMirror` and pass `XtermTerminal` in instead of `ITerminalInstance`.